types/options.d.ts

```diff
@@ -25,13 +25,18 @@ export type AsyncComponentPromise<Data=DefaultData<never>, Methods=DefaultMethod
   reject: (reason?: any) => void
 ) => Promise<Component | EsModuleComponent> | void;
 
-export type AsyncComponentFactory<Data=DefaultData<never>, Methods=DefaultMethods<never>, Computed=DefaultComputed, Props=DefaultProps> = () => {
-  component: AsyncComponentPromise<Data, Methods, Computed, Props>;
+export type AsyncComponentFactory<
+  Data = DefaultData<never>,
+  Methods = DefaultMethods<never>,
+  Computed = DefaultComputed,
+  Props = DefaultProps
+> = () => {
+  component: Promise<Component | EsModuleComponent>;
```

Comment:
Shouldn't component be typed?
```suggestion
  component: Promise<Component<Data, Methods, Computed, Props> | EsModuleComponent>;
```

We could also extend `EsModuleComponent` to accept the same generics